pkg/controller/secretannotator/azure/reconciler.go

```diff
@@ -86,6 +89,37 @@ func (r *ReconcileCloudCredSecret) Reconcile(request reconcile.Request) (reconci
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileCloudCredSecret) validateCloudCredsSecret(secret *corev1.Secret) error {
+
+	if _, ok := secret.Data[azureClientID]; !ok {
+		r.Logger.Errorf("Couldn't fetch key containing %v from cloud cred secret", azureClientID)
+		return r.updateSecretAnnotations(secret, constants.InsufficientAnnotation)
+	}
+
+	if _, ok := secret.Data[azureClientSecret]; !ok {
+		r.Logger.Errorf("Couldn't fetch key containing %v from cloud cred secret", azureClientSecret)
+		return r.updateSecretAnnotations(secret, constants.InsufficientAnnotation)
+	}
+
+	if _, ok := secret.Data[azureTenantID]; !ok {
+		r.Logger.Errorf("Couldn't fetch key containing %v from cloud cred secret", azureTenantID)
+		return r.updateSecretAnnotations(secret, constants.InsufficientAnnotation)
+	}
+
+	if _, ok := secret.Data[azureSubscriptionID]; !ok {
+		r.Logger.Errorf("Couldn't fetch key containing %v from cloud cred secret", azureSubscriptionID)
+		return r.updateSecretAnnotations(secret, constants.InsufficientAnnotation)
+	}
+
+	r.Logger.Info("Platform is azure: allowing to mint new credentials")
```

Comment:
can we put a TODO here that we should find a way to dynamically check whether these creds really can be used for minting (or passthrough) or whether they are useless (ie InsufficientAnnotation)?

Reply:
TODO note added